### DISCUSSION TITLE
Getting warning : x-for key cannot be an object, it must be a string or an integer

### DISCUSSION THREAD

**Discussion by abbasmashaddy72** [alpinejs/alpine#2381] [1. Help]
Getting a warning : x-for key cannot be an object, it must be a string or an integer

![Screenshot from 2021-11-16 17-14-25](https://user-images.githubusercontent.com/55267488/141979562-b6ccf4bb-6457-460c-bd38-f057dce66f59.png)

@rhengles

**Comment by rhengles** [comment]
Hello! If your `option` object has a property with a unique ID (for example, `option.id`), I suggest you use that in your key attribute on the template tag:
```html
<template x-for"..." :key="option.id">
```
Or else, you can just use the index:
```html
<template x-for"..." :key="index">
```

**Reply by abbasmashaddy72** [reply]
Okay I will check it out

Thanks

**Comment by SimoTod** [comment]
Assuming option.text is unique, you can use that one.

**Reply by abbasmashaddy72** [reply]
Okay I will check it out

Thanks

**Comment by abbasmashaddy72** [comment]
I was unable to fix it

I am using the component from [tailwind components](https://tailwindcomponents.com/component/multi-select-alpine-js)

I also want to pass the already selected option as selected but when I'm trying to do it is shown an error to convert the array to string and when I fix it not showing the options.

You find my code in [My Git repo](https://github.com/abbasmashaddy72/caringummah/blob/main/resources/views/forms/ummah_ea.blade.php)
It is available with all the factory and seeders build with Laravel.

If you want to Install the application and check then URL will be http://yourlocalhost/ummah/edit/{id}

Where you can find that family member option with multiple select option at the bottom of the screen.

**Reply by alex1897** [reply]
If still where not able to fix it, change the :key="option" to :key="option.value". The value should be unique.

**Reply by abbasmashaddy72** [reply]
Okay I will try

Thanks for the reply
